initial approach for new protocol

jodahoney opened the pull request:
Example of subject mr_session_report post upload (note history of xnat subject id and scan site):
https://ncanda.sri.com/redcap/redcap_v13.5.4/DataEntry/index.php?pid=20&id=D-00201-F-3&event_id=718&page=mr_session_report

Main component of this approach revolves around grabbing all projects that the subject has been uploaded under in xnat, first trying the default, and then if that fails trying the second one.

kipohl commented:
@jodahoney - nice - the variable site_map should be defined under global variable at the beginning of file - even better it should be defined in the configuration file as it is used across several scripts but leave it up to you

kipohl commented:
maybe define in /fs/ncanda-share/operations/sibis_sys_config.yml

jodahoney commented:
Updated:
- Site map now in global variables section of script
- Changed the logic so that if visit_site is empty in visit notes and we fail to get a scan under the default parameters, throw an error
    -  Example error that is created: https://github.com/sibis-platform/ncanda-operations/issues/15785

jodahoney commented:
@kipohl yes good question, if the subject is not found in Redcap it never reaches that point, 
```
ncanda@joe-pipeline-back[joe-pipeline_back_1]:~$ python -m pdb /sibis-software/ncanda-data-integration/scripts/redcap/import_mr_sessions -v --study-id X-99999-9-9
> /sibis-software/ncanda-data-integration/scripts/redcap/import_mr_sessions(8)<module>()
-> from __future__ import print_function
(Pdb) b 923
Breakpoint 1 at /sibis-software/ncanda-data-integration/scripts/redcap/import_mr_sessions:923
(Pdb) c
Namespace(event=None, force_update=False, force_update_stroop=False, max_days_after_visit=120, missing_only=False, no_stroop=False, no_upload=False, pipeline_root_dir=None, post_to_github=False, run_pipeline_script=None, site=None, study_id='X-99999-9-9', time_log_dir=None, verbose=True)
ERROR: study_id X-99999-9-9 not found in redcap!
The program exited via sys.exit(). Exit status: 1
```

jodahoney commented:
@kipohl the only flaw that I see in this setup is that for a visit that has taken place in the past that was a switched site visit but there was no scan, we don't have a way of sorting out which site handled the visit. 
One resolution that I think we talked about was just filling in the site changes noted in special_cases. Do you want me to do that before I merge with production?

jodahoney commented:
Removed mrireport_site check in the case where visit site and mri site exist. There are instances where some participants conduct their scan at one site, but complete their interviews and surveys through another site. We are designating visit site based on where a subject did their scan, so it sort of got in the way of that.
Moved it so the check is only done in the case where there is a visit_site but no scan site, so at least there is some check when no scan is collected and those visits should only be done at a single site.
